runtime/vm/vmprops.c

```diff
@@ -1433,3 +1445,651 @@ getMUtf8String(J9JavaVM *vm, const char *userString, UDATA stringLength)
+			if (!envSpace) {
+				strings[propIndex++] = "\\";
+			} else {
+				j9sysinfo_get_env("WINDIR", envSpace, envSize);
+				strings[propIndex++] = envSpace;
+			}
+		}
+	}
+
+#if JAVA_SPEC_VERSION < 17
+	/* Get the directory where the executable was started */
+	strings[propIndex++] = "user.dir";
+	if (0 == GetCurrentDirectoryW(EsMaxPath, unicodeTemp)) {
+		strings[propIndex++] = "\\";
+	} else {
+		convertToUTF8(PORTLIB, unicodeTemp, userdir, EsMaxPath);
+		strings[propIndex++] = userdir;
+	}
+#endif /* JAVA_SPEC_VERSION < 17 */
+
+	if (JAVA_SPEC_VERSION < 12) {
+		/* Get the timezone */
+		strings[propIndex++] = "user.timezone";
+		strings[propIndex++] = "";
+	}
+
+	result = createSystemPropertyList(env, strings, propIndex);
+	if (tempdir) {
+		j9mem_free_memory(tempdir);
+	}
+	if (envSpace) {
+		j9mem_free_memory(envSpace);
+	}
+	return result;
+}
+
+#else /* defined(WIN32) */
+
+jobject
+getPlatformPropertyList(JNIEnv *env, const char *strings[], int propIndex)
+{
+	PORT_ACCESS_FROM_ENV(env);
+	char *charResult = NULL;
+	char *envSpace = NULL;
+	jobject plist = NULL;
+#if JAVA_SPEC_VERSION < 17
+	char userdir[EsMaxPath] = {0};
+#endif /* JAVA_SPEC_VERSION < 17 */
+	char home[EsMaxPath] = {0};
+	char *homeAlloc = NULL;
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+	/* Hard coded file/path separators and other values */
+
+#if defined(J9ZOS390)
+	if (J2SE_VERSION_FROM_ENV(env)) {
+		strings[propIndex++] = "platform.notASCII";
+		strings[propIndex++] = "true";
+
+		strings[propIndex++] = "os.encoding";
+		strings[propIndex++] = "ISO8859_1";
+	}
+#endif
+
+#if JAVA_SPEC_VERSION < 17
+	strings[propIndex++] = "file.separator";
+	strings[propIndex++] = "/";
+
+	strings[propIndex++] = "line.separator";
+	strings[propIndex++] = "\n";
+
+	/* Get the directory where the executable was started */
+	strings[propIndex++] = "user.dir";
+	charResult = getcwd(userdir, EsMaxPath);
+	if (NULL == charResult) {
+		strings[propIndex++] = ".";
+	} else {
+		strings[propIndex++] = charResult;
+	}
+#endif /* JAVA_SPEC_VERSION < 17 */
+
+	strings[propIndex++] = "user.home";
+	charResult = NULL;
+#if defined(J9ZOS390)
+	charResult = getenv("HOME");
+	if (NULL != charResult) {
+		strings[propIndex++] = charResult;
+	} else {
+		uid_t uid = geteuid();
+		if (0 != uid) {
+			struct passwd *userDescription = getpwuid(uid);
+			if (NULL != userDescription) {
+				charResult = userDescription->pw_dir;
+				strings[propIndex++] = charResult;
+			}
+		} else {
+			char *loginID = getlogin();
+			if (NULL != loginID) {
+				struct passwd *userDescription = getpwnam(loginID);
+				if (NULL != userDescription) {
+					charResult = userDescription->pw_dir;
+					strings[propIndex++] = charResult;
+				}
+			}
+		}
+	}
+
+	/* there exist situations where one of the above calls will fail.  Fall through to the Unix solution for those cases */
+#endif
```

Comment:
Please fold the comment, start sentences with a capital and finish with a period.
The comment is missing on line 1818.